hw/ip/otp_ctrl/data/otp_ctrl.hjson

```diff
@@ -513,58 +494,130 @@
+
+    //////////////////////////////////////
+    // Integrity and Consistency Checks //
+    //////////////////////////////////////
+    { name: "CHECK_TRIGGER_REGWEN",
       desc: '''
-            Register write enable for !!INTEGRITY_CHECK_PERIOD_MSB and !!CONSISTENCY_CHECK_PERIOD_MSB.
+            Register write enable for !!CHECK_TRIGGER.
             ''',
       swaccess: "rw1c",
       hwaccess: "hro",
       fields: [
         { bits:   "0",
           desc: '''
-          When true, !!INTEGRITY_CHECK_PERIOD_MSB and !!CONSISTENCY_CHECK_PERIOD_MSB registers cannot be written anymore.
+          When cleared to 0, the !!CHECK_TRIGGER register cannot be written anymore.
+          Write 1 to clear this bit.
           '''
           resval: 1,
         },
       ]
     },
-    { name: "INTEGRITY_CHECK_PERIOD_MSB",
+    { name: "CHECK_TRIGGER",
+      desc: "Command register for direct accesses.",
+      swaccess: "r0w1c",
+      hwaccess: "hro",
+      hwqe:     "true",
+      regwen:   "CHECK_TRIGGER_REGWEN",
+      fields: [
+        { bits: "0",
+          name: "INTEGRITY",
+          desc: '''
+          Writing 1 to this bit triggers an integrity check. SW should monitor !!STATUS.CHECK_PENDING
+          and wait until the check has been completed. If there are any errors, those will be flagged
+          in the !!STATUS and !!ERR_CODE registers, and via the interrupts and alerts.
+          '''
+        }
+        { bits: "1",
+          name: "CONSISTENCY",
+          desc: '''
+          Writing 1 to this bit triggers a consistency check. SW should monitor !!STATUS.CHECK_PENDING
+          and wait until the check has been completed. If there are any errors, those will be flagged
+          in the !!STATUS and !!ERR_CODE registers, and via interrupts and alerts.
+          '''
+        }
+      ]
+    },
+    { name: "CHECK_REGWEN",
+      desc: '''
+            Register write enable for !!INTEGRITY_CHECK_PERIOD and !!CONSISTENCY_CHECK_PERIOD.
+            ''',
+      swaccess: "rw1c",
+      hwaccess: "hro",
+      fields: [
+        { bits:   "0",
+          desc: '''
+          When cleared to 0, !!INTEGRITY_CHECK_PERIOD and !!CONSISTENCY_CHECK_PERIOD registers cannot be written anymore.
+          Write 1 to clear this bit.
+          '''
+          resval: 1,
+        },
+      ]
+    },
+    { name: "CHECK_TIMEOUT",
+      desc: '''
+            Timeout value for the integrity and consistency checks.
+            ''',
+      swaccess: "rw",
+      hwaccess: "hro",
+      regwen:   "CHECK_REGWEN",
+      fields: [
+        { bits: "31:0",
+          desc: '''
+          Timeout value in cycles for the for the integrity and consistency checks. If an integrity or consistency
+          check does not complete within the timeout window, an error will be flagged in the !!STATUS register,
+          an otp_error interrupt will be raised, and an otp_check_failed alert will be sent out. The timeout should
```

Comment:
When both alert and interrupt triggered, will alert usually have higher priority? If that is the case, do we need to trigger both alert and interrupt here?

Reply:
Good point. Typically, the alerts are more severe - but due to the configurability of interrupts and the alert handler it is to some extent up to SW to decide the actual priority. 

The interrupt is triggered for consistency reasons (i.e., all errors lead to interrupts).
But we can discuss whether it makes sense to send out an interrupt at all if we anyway send out an alert for certain errors.

@tjaychen @cdgori, WDYT?

Reply:
i think it's better to keep the hardware simpler and just blast both out. We then leave it up to software to decide for their particular mode of operation, whether they want to be interrupted or alerted.

That's my thinking at least.

Reply:
yeah that is what I have been thinking as well...

Reply:
I could see either way as valid - but I think it ultimately depends on how we configure the alert escalation mechanisms.  If they are slow to accumulate, then an interrupt + alert for an error like this makes sense.  But if this alert will rapidly (or immediately) escalate to an interrupt then probably the interrupt is overkill and just wastes time in SW doing triage on multiple interrupt sources.

I feel like we will need to put some cycles into this alert/escalation policy soon-ish to resolve these types of things.  But I guess in lieu of that, lets just do alert + interrupt for now?

Reply:
Thanks for all the input. 
I agree that this will keep the HW simpler as Tim mentioned. And to this question, my initial thought was - "will it give SW potential redundant logic to deal with one error reported in two different forms". But I think Chris resolved my concern. I can see the comb of "alert with large accum_threshold + interrupt" being useful. Like Chris mentioned, would be good to put some policy on this.